### PR TITLE
util: basic config manager

### PR DESCRIPTION
**Pull request by hsirkar**
Usage:

```
import pipit as pp

pp.set_option("notebook_url", "https://localhost:8080")
pp.reset_option("notebook_url")

pp.get_option("log_level")
```

### CODE REVIEW

File: pipit/config.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2022-2023 Parallel Software and Systems Group, University of
+# Maryland. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+
+# Private dict that stores global config values
+_config = {
```
Comment (ocnkr) [inline pipit/config.py]: This looks good for a basic config manager. Here is a draft implementation in Hatchet: https://github.com/hatchet/hatchet/pull/511/files. It's similar to Pandas `set_options`. `registered_options` stores the default settings and `global_config` stores the current settings. The `get_option`, `set_option`, and `reset_option` functions are straightforward. There are also some validators to check if the given inputs are correct but I think the validator implementation has not finished yet.

Comment (hsirkar) [reply pipit/config.py]: Done

File: pipit/__init__.py [hunk 2]
```diff
@@ -4,3 +4,4 @@
 # SPDX-License-Identifier: MIT
 
 from .trace import Trace  # noqa: F401
+from .config import get_option, set_option  # noqa: F401
```
Comment (ocnkr) [inline pipit/__init__.py]: We should add reset_option

File: pipit/tests/config.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import pipit as pp
+
```
Comment (ocnkr) [inline pipit/tests/config.py:8]: Is there a function in which we might want to use global config? For example, in Hatchet we have the `tree` function that prints the CCT. One of the parameters it takes is `colormap`. We want to change colormap globally using `set_option`. Do we have a function like this in Pipit? If yes, can we add a test that it works?

**Review by ocnkr — CHANGES_REQUESTED** [review]

File: pipit/tests/config.py (hunk 3 above)
Comment (hsirkar) [reply pipit/tests/config.py:8]: There is a `set_option` function that is exposed to users. I'm not sure if this is what you mean.

Comment (hsirkar) [reply pipit/tests/config.py:8]: In Hatchet, if we specify a `colormap` parameter for the `tree` function, does it set the `colormap` globally? Or just once, for that specific tree?

Comment (ocnkr) [reply pipit/tests/config.py:8]: No, it just changes the colormap for that call. If we want to change it globally, we have to use set_option.

Comment (hsirkar) [reply pipit/tests/config.py:8]: That makes sense, we should be able to do the same in Pipit for timeline, etc.

Comment (ocnkr) [reply pipit/tests/config.py:8]: Yes, I was asking if we could add a test for something like that.

Comment (hsirkar) [reply pipit/tests/config.py:8]: If I understand what you're getting at, the timeline function would say something like:

```
def timeline( ..., colormap = None):
    ...
    if colormap = None:
        colormap = pp.get_config("colormap")
    ...
```

We currently have tests for `pp.get_config`. I think a specific test for the code above would be in the timeline test? Please correct me if I'm misunderstanding anything.

Comment (ocnkr) [reply pipit/tests/config.py:8]: Hmm, I think that makes sense. Let's discuss this with @bhatele later.

**Review by bhatele — APPROVED** [review]

**Pull request merged**